src/Entity/Field.php

```diff
@@ -20,19 +20,46 @@ class Field
     private array     $objectProperties = [];
     private array     $enumValues       = [];
     private string    $format           = '';
+    private ?int      $minimum          = null;
+    private ?bool     $exclusiveMinimum = null;
+    private ?int      $maximum          = null;
+    private ?bool     $exclusiveMaximum = null;
+    private ?int      $minLength        = null;
+    private ?int      $maxLength        = null;
+    private ?string   $pattern          = null;
+    private ?int      $minItems         = null;
+    private ?int      $maxItems         = null;
 
     public function __construct(
         string $name,
         FieldType $type,
         string $referenceName,
         bool $required,
-        bool $nullable
+        bool $nullable,
+        ?int $minimum,
+        ?bool $exclusiveMinimum,
+        ?int $maximum,
+        ?bool $exclusiveMaximum,
+        ?int $minLength,
+        ?int $maxLength,
+        ?string $pattern,
+        ?int $minItems,
+        ?int $maxItems
     ) {
-        $this->name          = $name;
-        $this->type          = $type;
-        $this->referenceName = $referenceName;
-        $this->required      = $required;
-        $this->nullable      = $nullable;
+        $this->name             = $name;
+        $this->type             = $type;
+        $this->referenceName    = $referenceName;
+        $this->required         = $required;
+        $this->nullable         = $nullable;
+        $this->minimum          = $minimum;
```

Comment:
Thanks for the awesome work! This validation is something that we needed for sure.
However, structurally I really don't like having like ten more optional parameters in the constructor. Could we encapsulate them in some sort of a value object, like it was done for FieldType?

Reply:
sure, will do

Reply:
done, please check